x/programs/cmd/simulator/src/lib.rs

```diff
@@ -296,9 +331,20 @@ where
         self.writer.write_all(b"'\n")?;
         self.writer.flush()?;
 
-        let resp = self.responses.next().ok_or(ClientError::Eof)??;
+        self.responses.next().ok_or(ClientError::Eof)?
+    }
 
-        Ok(resp)
+    pub fn run_step<T>(
+        &mut self,
+        caller_key: &str,
+        step: &Step,
+    ) -> Result<PlanResponseTyped<T>, ClientError>
```

Comment:
I think I would add a new error type here called a `StepError`:
```rust
#[derive(this_error::Error)]
enum StepError {
    #[error("{0}"]
    Client(#[from] ClientError),
    #[error("{0}"]
    Deserialization(#[from]borsh::io::Error),
}
```

Reply:
(please do this before I approve)

Reply:
Thanks, I added a new `Serialization` error as well since the step serialization is faillible in the new `StepError` enum.